numpy/lib/index_tricks.py

```diff
@@ -596,6 +597,425 @@ def __iter__(self):
     next = __next__
 
 
+class ndrange(_nx.collections_abc.Sequence):
+    """
+    class ndrange(stop)
+    class ndrange(start, stop[, step])
+
+    An N-dimensional range-like object for indexing arrays.
+
+    The ndrange type represents an immutable sequence of numbers and can be
```

Comment:
It doesn't represent a "sequence of numbers", it represents a "sequence of tuples of numbers"

Reply:
thanks.

Reply:
Does it represent a sequence of tuples, or an ndarray of tuples? I think that's an important design discussion - are we:
* generalizing `arange` to `ndrange` as `range` is to `xrange`
* Building a completely independent ND generalization of `range` that does not try to mirror `ndarray`

Reply:
I always had in mind the second option:

> Building a completely independent ND generalization of xrange that does not try to mirror ndarray